Source/Runtime/EmscriptenIntrinsics.cpp

```diff
@@ -224,6 +224,17 @@ namespace Runtime
 	{
 		return ungetc(character,vmFile(file));
 	}
+	DEFINE_INTRINSIC_FUNCTION4(env,_fread,_fread,I32,I32,pointer,I32,size,I32,count,I32,file)
+	{
+		if(pointer + uint64(size) * (count + 1) > instanceAddressSpaceMaxBytes)
```

Comment:
`count` needs to be extended to 64-bits before adding 1 to ensure it doesn't wrap around to 0.